### PR TITLE
Fix/add missing json interfaces

### PR DESCRIPTION
**Pull request by Magnus-Kuhn**

### CODE REVIEW

File: src/modules/attributes/local/AttributeSuccessorParams.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { ICoreDate, ICoreId } from "@nmshd/transport"
+import { ILocalAttribute, LocalAttributeJSON } from "./LocalAttribute"
+
+export type IAttributeSuccessorParams = Omit<ILocalAttribute, "id" | "createdAt"> & {
+    id?: ICoreId
+    createdAt?: ICoreDate
+}
+
+export type AttributeSuccessorParamsJSON = Omit<LocalAttributeJSON, "id" | "createdAt"> & {
```
Comment (slavistan) [inline src/modules/attributes/local/AttributeSuccessorParams.ts:9]: There's a bug.

**Review by slavistan — DISMISSED** [review]

**Comment by jkoenig134** [comment]
These JSON isn't used anywhere. What do you need it for?

File: src/modules/attributes/local/AttributeSuccessorParams.ts (hunk 1 above)
Comment (slavistan) [reply src/modules/attributes/local/AttributeSuccessorParams.ts:9]: (War ne Testnachricht)

**Comment by slavistan** [comment]
Shall be used in the runtime, instead of the naked `interface`s. As Thomas told me, no non-JSON interfaces should be imported into the runtime library at all.

**Comment by slavistan** [comment]
@jkoenig134 Good to go?

**Comment by jkoenig134** [comment]
But if the method doesn't accept These params as json, so it cant be used. And there is no serializable class to check it.

**Comment by slavistan** [comment]
The methods in the runtime will be changed to use the JSON interfaces. We have a problem with the schema generation, which bugs out for reasons unclear to me. Thomas got rid of the problem by removing imported non-json interfaces. I was told to use the JSON-interfaces (== DTOs) instead and that non-json interfaces should be used at all in the runtime.

**Comment by jkoenig134** [comment]
Ok then these types should be in the Runtime.

**Comment by jkoenig134** [comment]
I think the main question is why we are using that type at all in the runtime. The requests and responses of usecases should be de-coupled as strong as possible from the other libs.

**Review by stnmtz — APPROVED** [review]

**Pull request merged**